src/Symfony/Component/FeatureFlag/FeatureRegistry.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\FeatureFlag;
+
+use Symfony\Component\FeatureFlag\Exception\FeatureNotFoundException;
+
+final class FeatureRegistry implements FeatureRegistryInterface
+{
+    /**
+     * @param array<string, (\Closure(): mixed)> $features
+     */
+    public function __construct(private readonly array $features)
+    {
+    }
+
+    public function has(string $id): bool
+    {
+        return \array_key_exists($id, $this->features);
+    }
+
+    public function get(string $id): callable
+    {
+        return $this->features[$id] ?? throw new FeatureNotFoundException(sprintf('Feature "%s" not found.', $id));
+    }
+
+    public function getNames(): array
+    {
+        return array_keys($this->features);
```

Comment:
If I remember correctly. `id` is because of ContainerInterface. Otherwise it is `name`

Reply:
Yes, it was the main reason but it's not true since we removed the `ContainerInterface` usage. See https://github.com/symfony/symfony/pull/53213#discussion_r1554290804.